metadata/log_metadata.py

```diff
@@ -0,0 +1,226 @@
+        print("{} is not a valid process type. Logged anyways.".format(proc_type))
+
+    # Read the existing content if the log file exists
+    json_data, proc_data = None, None
+    if os.path.isfile(log_file):
+        with open(log_file, "r") as f:
+            json_data = json.load(f)
+        proc_data = json_data.get(proc_type, None)
+    # File doesn't exist yet
+    else:
+        json_data = {}
+
+    if proc_data is None:
+        proc_data = {}
+        json_data[proc_type] = proc_data
+
+    for k, v in metadata_dict.items():
+        if type(v) is str:
+            v = convert_to_numeric(v)
+
+        # Key doesn't exists yet
+        if k not in proc_data.keys():
+            proc_data[k] = v
+            continue
+
+        # Key already exists
+        if k in proc_data.keys():
+            k_count = sum([1 for cur_k in proc_data.keys() if k in cur_k])
+
+            # Key has only been added once before (i.e. primary value)
+            # Duplicate and add _1 postfix
+            if k_count == 1:
+                proc_data["{}_1".format(k)] = proc_data[k]
+
+                # Add new value
+                proc_data["{}_2".format(k)] = v
+            # Several keys already exists, just add additional with count postfix
+            else:
+                # Don't need a +1 as the count includes the primary value
+                proc_data["{}_{}".format(k, k_count)] = v
+
+            # Some additional values are required to be added to the existing
+            # value (e.g. run_time)
+            if k in metaconst_to_add:
+                if type(v) is not int and type(v) is not float:
+                    print(
+                        "Unsupported metadata value type for addition. "
+                        "Check metadata values. "
+                        "Value {} for key {} not added.".format(v, k),
+                        file=sys.stderr,
+                    )
+                    continue
+                else:
+                    proc_data[k] = proc_data[k] + v
+
+    # Write the json
+    with open(log_file, "w") as f:
+        json.dump(json_data, f)
+
+
+def main(args):
+    # This should come from constants
+    log_dir = os.path.join(args.sim_dir, "ch_log", LOG_FILENAME)
+
+    metadata_dict = getattr(args, METADATA_VALUES)
+
+    # Determine run_time from start and end time
+    if (
+        MetaConst.start_time.value in metadata_dict.keys()
+        and MetaConst.end_time.value in metadata_dict.keys()
+    ):
+        tdelta = datetime.strptime(
+            metadata_dict[MetaConst.end_time.value], TIMESTAMP_FMT
+        ) - datetime.strptime(metadata_dict[MetaConst.start_time.value], TIMESTAMP_FMT)
+        metadata_dict[MetaConst.run_time.value] = tdelta.total_seconds() / 3600
+
+    # Load the params
+    params = utils.load_sim_params(
+        os.path.join(args.sim_dir, "sim_params.yaml"), load_vm=True
+    )
+
+    # params metadata for LF
+    if args.proc_type == ProcTypeConst.LF.value:
```

Comment:
TODO: Put these ProcTypeConst in ?qcore

Reply:
Yeah good point, this will be done in a follow up PR.